core/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -667,4 +522,163 @@ public String toString() {
                 '}';
     }
 
+
+    final class RecoveryOutputStream extends OutputStream {
+        private final StoreFileMetaData md;
+        private final AtomicLong bytesSinceLastPause;
+        private final Translog.View translogView;
+        private long position = 0;
+
+        RecoveryOutputStream(StoreFileMetaData md, AtomicLong bytesSinceLastPause, Translog.View translogView) {
+            this.md = md;
+            this.bytesSinceLastPause = bytesSinceLastPause;
+            this.translogView = translogView;
+        }
+
+        @Override
+        public final void write(int b) throws IOException {
+            write(new byte[]{(byte) b}, 0, 1);
+        }
+
+        @Override
+        public final void write(byte[] b, int offset, int length) throws IOException {
+            sendNextChunk(position, new BytesArray(b, offset, length), md.length() == position + length);
```

Comment:
well this is just to pass on the buffer so I guess it's alwyas the buffer size ?